apps/freeablo/faworld/player.cpp

```diff
@@ -54,7 +54,26 @@ namespace FAWorld
     void Player::initCommon()
     {
         mWorld.registerPlayer(this);
-        mInventory.equipChanged.connect([this]() { updateSprites(); });
+        mInventory.inventoryChanged.connect([this](BasicInventory const& inventory, Item const& removed, Item const& added) {
+            (void)inventory;
+            (void)removed;
+
+            // Update player graphics.
+            updateSprites();
+
+            if (!added.isEmpty())
+            {
+                // Play inventory place sound.
```

Comment:
If inventories knew what type they were this would be much cleaner and wouldn't require the additional `cursorItemChanged` signal.
e.g.
```
switch (inventory.getType())
{
case InventoryType::body:
case InventoryType::leftHand:
case InventoryType::rightHand:
    // Update player graphics.
    updateSprites();
    break;
}

if (!added.isEmpty() /* && !silent && player == currentPlayer */ )
{
    // Play inventory place/grab sound.
    switch (inventory.getType())
    {
    case InventoryType::cursor:
        Engine::ThreadManager::get()->playSound("sfx/items/invgrab.wav");
        break;
    default:
        std::string soundPath = added.getInvPlaceSoundPath();
        Engine::ThreadManager::get()->playSound(soundPath);
        break;
    }
}
```
I might have a look at doing this refactor as well